design/backup_cancellation.md

```diff
@@ -0,0 +1,154 @@
+Additionally, the backup deletion controller prevents running backups from being deleted.
+
+
+## Goals
+- Enable users to cancel running backups through a `cancel` field in the backup specification
+- Cleanly cancel all associated async operations (BackupItemAction operations, DataUploads, PodVolumeBackups)
+- Provide clear backup phase transitions (InProgress → Cancelling → Cancelled)
+
+## Non Goals
+- Cancelling backups that have already completed or failed
+- Rolling back partially completed backup operations
+- Implementing cancellation for restore operations (future work)
+
+
+## High-Level Design
+The solution introduces a new `cancel` boolean field to the backup specification that users can set to `true` to request cancellation.
+Existing controllers (backup_controller, backup_operations_controller, backup_finalizer_controller) will check for this field and transition the backup to a `Cancelling` phase before returning early from their reconcile loops.
+
+A new dedicated backup cancellation controller will watch for backups in the `Cancelling` phase and coordinate the actual cancellation work.
+This controller will call `Cancel()` methods on all in-progress BackupItemAction operations (which automatically handles DataUpload cancellation), directly cancel PodVolumeBackups by setting their cancel flags, and finally transition the backup to `Cancelled` phase.
+The design uses a 5-second ticker to prevent API overload and ensures clean separation between cancellation detection and execution.
+
+## Detailed Design
+
+### API Changes
+Add a new field to `BackupSpec`:
+```go
+type BackupSpec struct {
+    // ... existing fields ...
+    
+    // Cancel indicates whether the backup should be cancelled.
+    // When set to true, Velero will attempt to cancel all ongoing operations
+    // and transition the backup to Cancelled phase.
+    // +optional
+    Cancel *bool `json:"cancel,omitempty"`
+}
+```
+
+Add new backup phases to `BackupPhase`:
+```go
+const (
+    // ... existing phases ...
+    BackupPhaseCancelling BackupPhase = "Cancelling" 
+    BackupPhaseCancelled  BackupPhase = "Cancelled"
+)
+```
+
+### Controller Changes
+
+#### Existing Controllers
+Modify `backup_controller.go`, `backup_operations_controller.go`, and `backup_finalizer_controller.go` to check for cancellation:
+```go
+// Early in each Reconcile method
+if backup.Spec.Cancel != nil && *backup.Spec.Cancel {
+    if backup.Status.Phase != BackupPhaseCancelling && backup.Status.Phase != BackupPhaseCancelled {
+        backup.Status.Phase = BackupPhaseCancelling
+        // Update backup and return
+        return ctrl.Result{}, c.Client.Patch(ctx, backup, client.MergeFrom(original))
+    }
+    return ctrl.Result{}, nil // Skip processing for cancelling/cancelled backups
+}
+```
+In addition, the `backup_operations_controller.go` will have a periodic check around backup progress updates, rather than running every time progress is updated to reduce API load.
```

Comment:
Backup operations controller reconciles once every 10 seconds. It should be fine to check for cancellation here on every reconcile.